packages/domain/candidature/src/corriger/corrigerCandidature.behavior.ts

```diff
@@ -74,8 +75,17 @@ export async function corriger(
     throw new DateÉchéanceGarantiesFinancièresRequiseError();
   }
 
-  if (this.estNotifiée && !candidature.statut.estÉgaleÀ(this.statut)) {
-    throw new StatutNonModifiableAprèsNotificationError();
+  if (this.estNotifiée) {
+    if (!candidature.statut.estÉgaleÀ(this.statut)) {
+      throw new StatutNonModifiableAprèsNotificationError();
+    }
+    if (candidature.typeGarantiesFinancières) {
```

Comment:
J'avoue que j'aurais été plus explicite quite à ce que ça soit plus verbeux ;) 

```suggestion
    if ((candidature.typeGarantiesFinancières && !this.typeGf) || (!candidature.typeGarantiesFinancières && this.typeGf) || (candidature.typeGarantiesFinancières && !this.typeGf.estÉgaleÀ(candidature.typeGarantiesFinancières))) {
     throw new TypeGarantiesFinancièresNonModifableAprèsNotificationError()
     }
```

Reply:
cette condition est totalement illisible je trouve, je préfère couper en deux... en quoi c'est plus explicite ?

Reply:
ça explicite juste avec les 3 vraies conditions
candidature.typeGarantiesFinancières && !this.typeGf
!candidature.typeGarantiesFinancières && this.typeGf
candidature.typeGarantiesFinancières && !this.typeGf.estÉgaleÀ(candidature.typeGarantiesFinancières)